test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingControllerSpec.scala

```diff
@@ -832,6 +832,46 @@ class UndertakingControllerSpec
 
       "display the page" when {
 
+        "When Change of undertaking business title is available in" in {
```

Comment:
I think the test names ("When Change of undertaking business title is available in" & "When legend is displayed in fieldset component") need to be updated. I think you don't need both. Only one will do. There is nothing dynamic on the page so you can check for the title and the legend in the same test.

Reply:
makes sense -  title and legend in  same test  now  @habeeb01